### PR TITLE
✨(company-representative): Mock silo

### PR DESCRIPTION
**Pull request by Siphesihle05**
Added the mock API for CompanyRep feature.

### CODE REVIEW

File: libs/api/companyrep/api/feature/src/lib/companyrep.controller.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Controller, Get } from '@nestjs/common';
```
Comment (arneschreuder) [inline libs/api/companyrep/api/feature/src/lib/companyrep.controller.ts:1]: There are no controllers in the api. Just resolvers.

File: libs/api/companyrep/api/feature/src/lib/companyrep.module.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Module } from '@nestjs/common';
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
+import { CompanyrepService } from 'libs/api/companyrep/service/feature/src/lib/companyrep.service';
```
Comment (arneschreuder) [inline libs/api/companyrep/api/feature/src/lib/companyrep.module.ts]: Please use this convention: `CompanyRepresentative` not `Companyrep`

Comment (arneschreuder) [inline libs/api/companyrep/api/feature/src/lib/companyrep.module.ts]: Please use `@graduates/api/company-representative/....` as import.

Comment (arneschreuder) [inline libs/api/companyrep/api/feature/src/lib/companyrep.module.ts]: This is technically incorrect. You should bundle the service in its own feature using its own feature module and them import that either directly in the shell or make a feature module for the entire company-representative feature that bundles the api and service and repository layer.

File: libs/api/companyrep/api/feature/src/lib/companyrep.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    async companyrep(@Args('id') id : string): Promise<Companyrep>{
+        const example = await this.companyrepService.findOneById(id);
+        if(!example){
+            throw new NotFoundException(id);
+        }
+        return  example;
+    } 
+
+    @Mutation(returns => Companyrep)
+    async addCompanyrep(@Args('newCompanyrepData') newCompanyrepData: NewCompanyrepInput,): Promise<Companyrep> {
```
Comment (arneschreuder) [inline libs/api/companyrep/api/feature/src/lib/companyrep.resolver.ts:25]: Companyrep -> CompanyRepresentative and the API engineers need to standardise the conventions for create, fetch, delete and update

File: libs/api/companyrep/api/shared/data-access/src/lib/companyrep.model.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { ObjectType, Field, ID, Directive, Int } from "@nestjs/graphql";
+
+@ObjectType({description: 'companeyrep profile'})
```
Comment (arneschreuder) [inline libs/api/companyrep/api/shared/data-access/src/lib/companyrep.model.ts:3]: Please consult with the API engineers. The naming needs to be standardised. You can merge this in for now and fix later? But others have called this `*.entity.ts`

File: libs/api/companyrep/api/shared/data-access/src/lib/new-companyrep.input.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { Field, InputType } from '@nestjs/graphql';
+
+
+@InputType()
+export class NewCompanyrepInput {
+    @Field(type => String)
```
Comment (arneschreuder) [inline libs/api/companyrep/api/shared/data-access/src/lib/new-companyrep.input.ts:6]: This should technically be `type => ID` and force the types for the other fields as well please.

File: libs/api/companyrep/api/shared/data-access/src/lib/new-companyrep.input.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { Field, InputType } from '@nestjs/graphql';
```
Comment (arneschreuder) [inline libs/api/companyrep/api/shared/data-access/src/lib/new-companyrep.input.ts:1]: Once again, I think the API engineers need to standardise here. I wouldve called this CreateCompanyRepresentativeRequest

File: libs/api/companyrep/api/shared/data-access/src/lib/shared-data-access.module.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { Module } from '@nestjs/common';
+import { Companyrep } from './companyrep.model';
+import { NewCompanyrepInput } from './new-companyrep.input';
+
+@Module({
+  controllers: [],
+  providers: [NewCompanyrepInput, Companyrep],
```
Comment (arneschreuder) [inline libs/api/companyrep/api/shared/data-access/src/lib/shared-data-access.module.ts:7]: Not sure if GraphQL types are listed as providers. Please check?

File: libs/api/companyrep/service/feature/src/lib/companyrep.service.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common';
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
+import { Companyrep } from '../../../../api/shared/data-access/src/lib/companyrep.model';
```
Comment (arneschreuder) [inline libs/api/companyrep/service/feature/src/lib/companyrep.service.ts:3]: Please use the '@graduates/path' import notation.

Comment (arneschreuder) [reply libs/api/companyrep/service/feature/src/lib/companyrep.service.ts:3]: Also, you never import a specific file. Only barrel the module and extract from that what you need.

File: libs/api/companyrep/service/feature/src/lib/service-feature.module.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { Module } from '@nestjs/common';
+import { CompanyrepService } from './companyrep.service';
+
+@Module({
+  controllers: [],
+  providers: [CompanyrepService],
```
Comment (arneschreuder) [inline libs/api/companyrep/service/feature/src/lib/service-feature.module.ts:6]: Naming convention

File: libs/api/shell/feature/src/lib/api-shell-feature.module.ts [hunk 10]
```diff
@@ -1,13 +1,18 @@
 import { FeatureModule as ApiExampleFeatureModule } from '@graduates/api/example/api/feature';
 import { ApiStorageApiFeatureModule as ApiStorageFeatureModule } from '@graduates/api/storage/api/feature';
 import { Module } from '@nestjs/common';
-import {GraphQLModule} from '@nestjs/graphql'
+import { GraphQLModule } from '@nestjs/graphql';
+import { CompanyrepModule as ApiFeatureModule} from 'libs';
```
Comment (arneschreuder) [inline libs/api/shell/feature/src/lib/api-shell-feature.module.ts:5]: No, please call this something more descriptive. ApiCompanyRepApiFeatureModule similar to the others.

File: libs/api/shell/feature/src/lib/api-shell-feature.module.ts [hunk 11]
```diff
@@ -1,13 +1,18 @@
...
-  imports: [ApiExampleFeatureModule,
-  GraphQLModule.forRoot({
-    autoSchemaFile: true,
-    driver: ApolloDriver
+  imports: [
+    ApiExampleFeatureModule, 
+    ApiFeatureModule,
+    GraphQLModule.forRoot<ApolloDriverConfig>({
+      driver: ApolloDriver,
+      autoSchemaFile: 'schema.gql',
```
Comment (arneschreuder) [inline libs/api/shell/feature/src/lib/api-shell-feature.module.ts:14]: Perhaps just output this to a better location? Perhaps a library? There we can then generate interfaces from this graphql schema.

File: package.json [hunk 12]
```diff
@@ -34,6 +34,7 @@
     "@nestjs/core": "^8.0.0",
     "@nestjs/graphql": "^10.0.7",
     "@nestjs/platform-express": "^8.0.0",
+    "@nestjsi/class-validator": "^0.1.4",
```
Comment (arneschreuder) [inline package.json:37]: Please notify the others that dependencies have been added. They need to yarn install again

File: tsconfig.base.json [hunk 13]
```diff
@@ -40,6 +40,15 @@
         "libs/api/example/service/shared/interfaces/data-access/src/index.ts"
       ],
       "@graduates/api/shell/feature": ["libs/api/shell/feature/src/index.ts"],
+      "@graduates/api\\companyrep\\api/data-acces": [
```
Comment (arneschreuder) [inline tsconfig.base.json]: Slashes look odd. Please fix.

File: workspace.json [hunk 14]
```diff
@@ -11,6 +11,9 @@
     "api-example-service-feature": "libs/api/example/service/feature",
     "api-example-service-shared-interfaces-data-access": "libs/api/example/service/shared/interfaces/data-access",
     "api-shell-feature": "libs/api/shell/feature",
+    "api\\companyrep\\api-feature": "libs/api/companyrep/api/feature",
```
Comment (arneschreuder) [inline workspace.json]: Here as well.

**Review by arneschreuder — CHANGES_REQUESTED** [review]
Thank you very much for the PR. Almost there. A few small fixes.

**Comment by arneschreuder** [comment]
Please tag your "feature project" on the meta together with the labels.

**Comment by arneschreuder** [comment]
Also, changing a directory name like this: https://github.com/COS-301/graduates/pull/191/commits/7a9823621f0dc256c7a311922782ac8aec3a8b81 is not correct. You should use the nx console to rename projects. Directories are not just renamed. Please fix.

**Review by arulagrawal — DISMISSED** [review]
I tried to look for issues and in some cases offered a quick fix. perhaps @arneschreuder should look at this as well since this is our first PR?

**Comment by arulagrawal** [comment]
I just made that comment in the wrong tab :(. Is there anyway to delete?

**Comment by Siphesihle05** [comment]
> Also, changing a directory name like this: [7a98236](https://github.com/COS-301/graduates/commit/7a9823621f0dc256c7a311922782ac8aec3a8b81) is not correct. You should use the nx console to rename projects. Directories are not just renamed. Please fix.

I am using the the move commad in nx Console is that okay?

**Comment by arneschreuder** [comment]
> > Also, changing a directory name like this: [7a98236](https://github.com/COS-301/graduates/commit/7a9823621f0dc256c7a311922782ac8aec3a8b81) is not correct. You should use the nx console to rename projects. Directories are not just renamed. Please fix.
> 
> I am using the the move commad in nx Console is that okay?

If that is the case, then yes, it just looked weird in the PR. If you are doing this go ahead.

**Comment by arneschreuder** [comment]
Please resolve conflicts and update branch. This PR is getting stale.

**Pull request closed without merging**